les/serverpool.go

```diff
@@ -239,9 +242,9 @@ func (s *serverPool) start() {
 	unixTime := s.unixTime()
 	s.ns.ForEach(sfHasValue, nodestate.Flags{}, func(node *enode.Node, state nodestate.Flags) {
 		s.calculateWeight(node)
-		if n, ok := s.ns.GetField(node, sfiNodeHistory).(nodeHistory); ok && n.nextDial > unixTime {
-			wait := n.nextDial - unixTime
-			lastWait := n.nextDial - n.lastDial
+		if n, ok := s.ns.GetField(node, sfiNodeHistory).(nodeHistory); ok && n.redialWaitEnd > unixTime {
+			wait := n.redialWaitEnd - unixTime
+			lastWait := n.redialWaitEnd - n.redialWaitStart
 			if wait > lastWait {
```

Comment:
Can we simplify the logic as `wait := n.redialWaitEnd - unixTime`?

Reply:
The rest of the logic is just a protection against setting crazy high timeouts if the system clock was adjusted. In the normal case (if `unixTime > n.redialWaitStart`) it can't hurt anything so I think it should stay.